Microsoft.Toolkit.Uwp.UI.Controls/TextToolbar/TextToolbar.Events.cs

```diff
@@ -0,0 +1,318 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    using System;
+    using System.Collections.Specialized;
+    using Microsoft.Toolkit.Uwp.UI.Controls.TextToolbarButtons;
+    using Windows.System;
+    using Windows.UI.Core;
+    using Windows.UI.Xaml;
+    using Windows.UI.Xaml.Controls;
+    using Windows.UI.Xaml.Input;
+
+    /// <summary>
+    /// Toolbar for Editing Text attached to a RichEditBox
+    /// </summary>
+    public partial class TextToolbar
+    {
+        /// <summary>
+        /// Attaches/Detaches the Key events for Shortcuts
+        /// </summary>
+        /// <param name="obj">TextToolbar</param>
+        /// <param name="args">Property Changed Args</param>
+        public static void OnEditorChanged(DependencyObject obj, DependencyPropertyChangedEventArgs args)
+        {
+            var bar = obj as TextToolbar;
+            if (bar != null)
+            {
+                var oldEditor = args.OldValue as RichEditBox;
+                var newEditor = args.NewValue as RichEditBox;
+
+                if (oldEditor != null)
+                {
+                    oldEditor.RemoveHandler(KeyDownEvent, bar.KeyEventHandler);
+                }
+
+                if (newEditor != null)
+                {
+                    newEditor.AddHandler(KeyDownEvent, bar.KeyEventHandler, handledEventsToo: true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Checks if a Shortcut Combination is pressed, by going through the list of attached buttons.
+        /// </summary>
+        /// <param name="sender">RichEditBox</param>
+        /// <param name="e">Key args</param>
+        private void Editor_KeyDown(object sender, KeyRoutedEventArgs e)
+        {
+            LastKeyPress = e.Key;
+
+            var root = GetTemplateChild(RootControl) as CommandBar;
+            if (root != null)
+            {
+                if (ControlKeyDown && e.Key != VirtualKey.Control)
+                {
+                    if (e.Handled)
+                    {
+                        Editor.Document.Undo();
```

Comment:
This is to handle overrides of default shortcuts, such as CTRL + - for strikethrough. Since this shortcut appears to delete text by default, this is used to undo the delete and apply the strikethrough if there is a ToolbarButton that handles it, this is improved in me next commit.